app/src/main/java/ch/hikemate/app/ui/map/RunHikeScreen.kt

```diff
@@ -137,6 +241,55 @@ private fun RunHikeContent(hike: DetailedHike, navigationActions: NavigationActi
   }
 }
 
+/**
+ * Parses a location result and updates the user's position on the map.
+ *
+ * @param locationResult The location result to parse
+ * @param userLocationMarker The marker representing the previous user's location on the map
+ * @param mapView The map view where the user's location is displayed
+ * @param hike The hike where the user is running
+ */
+private fun parseLocationUpdate(
+    locationResult: LocationResult,
+    userLocationMarker: Marker?,
+    mapView: MapView,
+    hike: DetailedHike
+): Marker? {
```

Comment:
Warning: This function was pretty hard to read to be honest. I think this could help:
Instead of putting if and else and if... you could use some `guards ifs`:

```
if(no loc){
  clear, return null
}
...
if (no projection) {
  return null
}
...
// get location to center on
// only 1 MapUtils.updateUserPosition call here
```

Reply:
What do you mean with `guards ifs`? I don't know this?

Reply:
I've tried to simplify it